plugins/chat/assets/javascripts/discourse/services/chat-composer-warnings-tracker.js

```diff
@@ -0,0 +1,143 @@
+import Service, { inject as service } from "@ember/service";
+import { ajax } from "discourse/lib/ajax";
+import discourseDebounce from "discourse-common/lib/debounce";
+import { bind } from "discourse-common/utils/decorators";
+import { mentionRegex } from "pretty-text/mentions";
+import { cancel } from "@ember/runloop";
+import { tracked } from "@glimmer/tracking";
+
+const MENTION_RESULT = {
+  invalid: -1,
+  unreachable: 0,
+  over_members_limit: 1,
+};
+
+const MENTION_DEBOUNCE_MS = 1000;
+
+export default class ChatComposerWarningsTracker extends Service {
+  @service siteSettings;
+
+  // Track mention hints to display warnings
+  @tracked unreachableGroupMentions = [];
+  @tracked overMembersLimitGroupMentions = [];
+  @tracked tooManyMentions = false;
+  @tracked mentionsCount = 0;
+  @tracked mentionsTimer = null;
+
+  // Complimentary structure to avoid repeating mention checks.
+  _mentionWarningsSeen = {};
+
+  willDestroy() {
+    cancel(this.mentionsTimer);
+  }
+
+  @bind
+  trackMentions(message) {
+    this.message = message;
+
+    this.mentionsTimer = discourseDebounce(
+      this,
+      this._trackMentions,
+      MENTION_DEBOUNCE_MS
+    );
```

Comment:
Can't try this right now, but you should be able to do this:

```
this.mentionsTimer = discourseDebounce(
  this,
  this._trackMentions,
  this.message,
  MENTION_DEBOUNCE_MS
);
```

and `message` will now be passed a param to `_trackMentions`

Reply:
Aaaaah TIL. Thought those arguments were timeout specific. Will try it!